File: LatvijasPasts/Models/PersonalInfo.cs
﻿namespace LatvijasPasts.Models
{
    public class PersonalInfo
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string PhoneNumber { get; set; }
        public string Email { get; set; }
        public List<WorkExperience>? WorkExperiences { get; set; }
        public List<Education>? Educations { get; set; }
    }
}

File: Frontend Latvijas Pasts/latvijaspasts/app/services/personalinfo.ts
export interface CvRequest {
    firstName: string;
    lastName: string;
    phoneNumber: string;
    email: string;
    workExperiences: WorkExperience;
    educations: Education;
}

export const getAllCv = async () =>  {
    const response = await fetch("http://localhost:5108/cv");
    return response.json();
};

export const createCv = async (cvRequest: CvRequest) => {
    await fetch("http://localhost:5108/cv", {
        method: "POST",
        headers: {
            "content-type": "application/json",
        },
        body: JSON.stringify(cvRequest)
    });
};

    export const updateCv = async (id: number, cvRequest: CvRequest) => {
        await fetch(`http://localhost:5108/cv/${id}`, {
            method: "PUT",
            headers:{
                "content-type": "application/json",
            },
            body: JSON.stringify(cvRequest),
        });
    };

    export const deleteCv = async (id: number) => {
        await fetch(`http://localhost:5108/cv/${id}`, {
            method: "DELETE",
        });
    };